js/packages/ufg-client/test/unit/fetch-resources.spec.ts

```diff
@@ -72,6 +73,36 @@ describe('fetch-resource', () => {
     assert.deepStrictEqual(resource2, mockResource)
   })
 
+  it('fetch with concurrency limitation', async () => {
+    const mockResources = []
+    let count = 0
+    for (let i = 0; i < 10; i++) {
+      mockResources.push(
+        makeResource({
+          url: `http://something${i}`,
+        }),
+      )
+      nock(`http://something${i}`).get('/').once().reply(limitServerParallelRequests)
+    }
```

Comment:
A am still kinda skeptical about this `for` loop. I understand what you tried to do, but I would like us to avoid such usage in general, maybe it is fine for the tests.
Anyway, I want to recommend this, just to not give chance for this practice.

```suggestion
    const mockResources = Array({length: 10}, (_, index) => makeResource({url: `http://something/${index}`}))
    let count = 0
     nock('http://something').get(/\d+/).times(mockResources.length).reply(limitServerParallelRequests)
```

Reply:
Thank you for taking the time to mention that

Reply:
That actually, doesn't work, let's talk about it.